### ISSUE TITLE
Troubleshooting emacs hanging when replying to certain messages with org-msg

### ISSUE THREAD

**Issue by peterhoeg** [jeremy-compostella/org-msg#154]
I'm looking for assistance on how to trouble-shoot some issues with org-msg.

Certain mails will consistently hang emacs (it starts eating CPU with no way to interrupt it) when replying to them with org-msg. If I do a "regular" reply with mu4e-compose I have no issues. Some senders are more likely to trigger it than others but not not all mails from the offending senders will cause the hang.

I cannot use the profiler to see *where* it gets stuck as I have no way of interacting with emacs once it hangs.

**Comment by yantar92** [comment]
One way is setting debug-on-error to t and then sending SIGUSR2 signal to Emacs process when Emacs hangs. As a result, you should be able to get a backtrace of what was running during the hang.

**Comment by jeremy-compostella** [comment]
Do you have to comment out all of them or only some of them?

**Comment by jsilve24** [comment]
Yes, good question. An hour or so of debugging revealed that the problem seems to be in `message-font-lock-keywords` and `gnus-message-citation-keywords`. Moreover, it seems to be only in the part of these two that deal with font-lock of quote levels. See my prior comment where I give the section of `message-font-lock-keywords` that needs to be commented out to make that one work. `gnus-message-citation-keywords` on the other hand seems to have nothing in it other than code related to quote levels and I foudn that I had to pretty much comment out that entire thing to make it work (at least in my machine).

**Comment by jsilve24** [comment]
I don't understand org-msg well enough to know if this will work, but here it goes. 

```
From: Justin Silverman <myemail@psu.edu>
To: "Foo, Bar" <foo5431@psu.edu>
Subject: Re: Test
Date: Mon, 13 Feb 2023 21:19:50 -0500
In-reply-to: <CH0PR02MB824241D16A0F5E72DE51133DA2DD9@CH0PR02MB8242.namprd02.prod.outlook.com>
--text follows this line--
#+OPTIONS: html-postamble:nil num:nil ^:{} toc:nil author:nil email:nil \n:t tex:dvipng eval:nil d:nil
#+STARTUP: hidestars indent inlineimages
:PROPERTIES:
:reply-to: ("/tmp/mm-yi7MOB.html" "/tmp/cidYyAWvu")
:attachment: nil
:alternatives: (text html)
:END:


--citation follows this line (read-only)--
"Foo, Bar" <foo5431@psu.edu> writes:

Test



---

Foo Bar | Graduate Programs Assistant

E103/E105 Westgate Building | foo45612@psu.edu

University Park | PA | 16802

PSU_IST_PMS_287_284-300dp
```

**Comment by jeremy-compostella** [comment]
This is a very very simple reply so I got to ask would it have hang Emacs
if you did not commented out those font locks ?

**Comment by jsilve24** [comment]
Well S***%. Yes, that email exactly was my test case for most of today. But in the past 2 hours I just updated all my emacs packages and now that email is working fine. Maybe the bug got fixed upstream? Its been hanging around for months now... I will watch for it, if it doesn't show up in the near future I guess my part in this is fixed.

**Comment by jeremy-compostella** [comment]
That was going to be my next suggestion if you were producing with such a
simple email. I would hit the issue way too often to not investigate.

I remember running into some similar issues a very long time ago. I don't
remember if I had to fix it myself or not.

Jeremy

**Comment by peterhoeg** [comment]
Here are a few of my observations:

1. this happens almost exclusively with mails from a Spanish customer of ours (written in English) using Outlook. Keep in mind that I get mails from people typically writing in Chinese, Korean and Japanese from time to time *without* any issues.
2. I can only replicate the issue if I have used org-agenda first (which I almost always do).
3. I’ve just tried updating all emacs packages, no change.

**Comment by peterhoeg** [comment]
I haven't tried finding a definite case that triggers this, but I *think* it has to do with this scenario:

1. the customer in question has *very* long disclaimer at the bottom which renders on one line (about 1300 characters)
2. my signature ends up after the disclaimer

If I hit reply on that message, it's initially OK when the disclaimer is further down (I'm guessing emacs is being smart about font-locking and not doing it until it gets into view).
If I jump to the very end of the mail, I get the hang.
If I then manage to kick emacs enough times to get back control, disable font-locking, delete the signature and enable font-locking again, it's OK.

**Comment by jeremy-compostella** [comment]
Is there any portion of that signature reproducing the issue if pasted in a
org msg buffer that you could share?

**Comment by peterhoeg** [comment]
> Is there any portion of that signature reproducing the issue if pasted in a org msg buffer that you could share?

I need to carve out a little bit of time in order to come up with a case that reliably reproduces this.

**Comment by JonathanReeve** [comment]
I'm having a similar issue when replying to messages. For me, it's all messages, anad it doesn't matter whether I have any signatures set, or what my settings are in org-msg. I'm using Doom Emacs and I suspect there might be something going on there, unless you're not using Doom and you're getting the same issue. I [opened an issue here with Doom](https://github.com/doomemacs/doomemacs/issues/7133). Are you getting any freezes related to enabling/disabling word wrap?

**Comment by peterhoeg** [comment]
It definitely has to do with long lines. Will find more details.

**Comment by peterhoeg** [comment]
I have not had this issue since I started using this:
```
(add-hook 'org-msg-edit-mode-hook
          (lambda ()
            (setq org-font-lock-keywords (delq 'gnus-message-citation-keywords org-font-lock-keywords))))
```
